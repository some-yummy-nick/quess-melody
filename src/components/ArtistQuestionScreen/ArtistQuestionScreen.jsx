import React from "react";
import PropTypes from "prop-types";
import {GameType} from "../../constants";
import {connect} from "react-redux";

const ArtistQuestionScreen = ({question: {answers, song}, onAnswer, mistakes, maxMistakes, renderPlayer}) =>
    <>
        <h2 className="game__title">Кто исполняет эту песню?</h2>
        <div className="game__track">
            <div className="track">
                {renderPlayer(song, 0)}
            </div>
        </div>
        <form className="game__artist">
            {answers.map((answer, i) => {
                return (
                    <div key={`answer-${i}`} className="artist">
                        <input className="artist__input visually-hidden" type="radio" name="answer"
                               value={`answer-${i}`}
                               id={`answer-${i}`}
                               onChange={(evt) => {
                                   evt.preventDefault();
                                   onAnswer({answers, song}, answer, mistakes, maxMistakes);
                               }}
                        />
                        <label className="artist__name" htmlFor={`answer-${i}`}>
                            <img className="artist__picture" src={answer.picture} alt={answer.artist}/>
                            {answer.artist}
                        </label>
                    </div>
                );
            })}
        </form>
    </>;

ArtistQuestionScreen.propTypes = {
    onAnswer: PropTypes.func.isRequired,
    question: PropTypes.shape({
        answers: PropTypes.arrayOf(PropTypes.shape({
            artist: PropTypes.string.isRequired,
            picture: PropTypes.string.isRequired,
        })).isRequired,
        song: PropTypes.shape({
            artist: PropTypes.string.isRequired,
            src: PropTypes.string.isRequired,
        }).isRequired,
        type: PropTypes.oneOf([GameType.ARTIST, GameType.GENRE]).isRequired,
    }).isRequired,
    renderPlayer: PropTypes.func.isRequired,
};
export {ArtistQuestionScreen};

const mapStateToProps = state => ({
    mistakes: state.mistakes,
    maxMistakes: state.maxMistakes,
});

export default connect(
    mapStateToProps
)(ArtistQuestionScreen);
