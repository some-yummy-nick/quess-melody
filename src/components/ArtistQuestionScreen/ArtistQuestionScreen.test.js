import React from 'react';
import renderer from 'react-test-renderer';
import {ArtistQuestionScreen} from './ArtistQuestionScreen.jsx';

const question = {
    type: `artist`,
    answers: [{
        id: 1,
        picture: `picture`,
        artist: `John Snow`
    }],
    song: {
        id: 1,
        artist: `Some`,
        src: `src`
    }
};

it(`ArtistQuestionScreen correctly renders after relaunch`, () => {
    const handleClick = jest.fn();
    const renderPlayer = jest.fn();
    const tree = renderer
        .create(<ArtistQuestionScreen
            question={question}
            onAnswer={handleClick}
            renderPlayer={renderPlayer}
        />, {
            createNodeMock: () => {
                return {};
            }
        }).toJSON();
    expect(tree).toMatchSnapshot();
});
