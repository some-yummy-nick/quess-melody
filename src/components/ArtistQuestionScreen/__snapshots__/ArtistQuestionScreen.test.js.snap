// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistQuestionScreen correctly renders after relaunch 1`] = `
Array [
  <h2
    className="game__title"
  >
    Кто исполняет эту песню?
  </h2>,
  <div
    className="game__track"
  >
    <div
      className="track"
    />
  </div>,
  <form
    className="game__artist"
  >
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-1"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="answer-0"
      />
      <label
        className="artist__name"
        htmlFor="answer-1"
      >
        <img
          alt="John Snow"
          className="artist__picture"
          src="picture"
        />
        John Snow
      </label>
    </div>
  </form>,
]
`;
